### PR TITLE
Fix memory leak at osquery/devtools/shell.cpp

### LINKED ISSUE #2548: shell: Fix memory leak at osquery/devtools/shell.cpp:1551
Build osquery using:

1. `make deps`
2. `make sanitize -j 10`

Then invoke the shell using:
```bash
$ PATH=/usr/local/osquery/bin:$PATH ./build/linux/osquery/osqueryi -A osquery_info
+-------+--------------------+-------------+--------------+------------+----------------+--------------+------------+---------+
| pid   | version            | config_hash | config_valid | extensions | build_platform | build_distro | start_time | watcher |
+-------+--------------------+-------------+--------------+------------+----------------+--------------+------------+---------+
| 20200 | 1.8.2-162-gb77c217 |             | 0            | inactive   | ubuntu         | xenial       | 1475046797 | -1      |
+-------+--------------------+-------------+--------------+------------+----------------+--------------+------------+---------+

=================================================================
==20200==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 18 byte(s) in 1 object(s) allocated from:
    #0 0xcc396b in operator new[](unsigned long) /tmp/llvm-20160901-473558-1grod1w/llvm-3.8.1.src/projects/compiler-rt/lib/asan/asan_new_delete.cc:64:37
    #1 0xcc6f71 in osquery::launchIntoShell(int, char**) /home/teddy/git/osquery/osquery/devtools/shell.cpp:1551:17
    #2 0xcff116 in main /home/teddy/git/osquery/osquery/main/shell.cpp:184:15
    #3 0x7f1d7df7982f in __libc_start_main /build/glibc-GKVZIf/glibc-2.23/csu/../csu/libc-start.c:291

SUMMARY: AddressSanitizer: 18 byte(s) leaked in 1 allocation(s).
```

Appending the `PATH` allows `LSAN` to locate the `llvm-symbolizer` tool.

### PR DESCRIPTION
**Pull request by rogeliodh**
Fixes #2548

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by mtmcgrew** [comment]
ok to test

File: osquery/devtools/shell.cpp [hunk 1]
```diff
@@ -1548,10 +1548,10 @@ int launchIntoShell(int argc, char** argv) {
...
     // Helper meta commands from shell switches.
     std::string query = (FLAGS_L) ? ".tables" : ".all " + FLAGS_A;
-    char* cmd = new char[query.size() + 1];
-    memset(cmd, 0, query.size() + 1);
-    std::copy(query.begin(), query.end(), cmd);
-    rc = do_meta_command(cmd, &data);
+    boost::scoped_array<char> cmd(new char[query.size() + 1]);
+    memset(cmd.get(), 0, query.size() + 1);
+    std::copy(query.begin(), query.end(), cmd.get());
+    rc = do_meta_command(cmd.get(), &data);
```
Comment (theopolis) [inline osquery/devtools/shell.cpp]: Could we just `delete[] cmd` instead of using a `boost::scoped_array`?

Comment (rogeliodh) [inline osquery/devtools/shell.cpp]: yes, of course, that's the simplest way to do it, but it is also the most prone to more mistakes in the future. You pick.

Comment (theopolis) [inline osquery/devtools/shell.cpp]: I agree, in this case let's opt for the `delete[] cmd` since this file is already very C89/C-extension heavy. :P

Comment (rogeliodh) [inline osquery/devtools/shell.cpp]: I've update the pull request with your suggestion

**Comment by osqueryer** [comment]
:-1: The commit 21c1686bc5bf6fb711227b10712186b6f028d220 (Job results: [3408](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3408/)) failed one or more tests (**OS X/Linux**).

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
:-1: The commit 21c1686bc5bf6fb711227b10712186b6f028d220 (Job results: [3411](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3411/)) failed one or more tests (**OS X/Linux**).

**Comment by theopolis** [comment]
ok to test

**Comment by theopolis** [comment]
I'm provisioning another Ubuntu 14.04 hosts for the test. I'm confident this passes (on other Linux platforms) and that this is a test error.

**Comment by osqueryer** [comment]
:-1: The commit 21c1686bc5bf6fb711227b10712186b6f028d220 (Job results: [3412](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3412/)) failed one or more tests (**OS X/Linux**).

**Pull request merged**